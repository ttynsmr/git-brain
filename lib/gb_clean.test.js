const { cleanFilter } = require('./gb_clean');
const fs = require('fs');

describe('clean filter actual file', () => {
  function createTestActualFile() {
    fs.writeFileSync('.test.in', 'for example 222');
  }

  function removeTestActualFile() {
    try {
      fs.unlinkSync('.test.in');
    } catch (err) {}
    try {
      fs.unlinkSync('.test.out');
    } catch (err) {}
  }

  beforeEach(() => {
    createTestActualFile();
  });

  afterEach(() => {
    removeTestActualFile();
  });

  test('file stream', async () => {
    const reader = fs.createReadStream('.test.in', 'utf8');
    const writer = fs.createWriteStream('.test.out', 'utf8');
    await cleanFilter(reader, writer);
    let out = fs.readFileSync('.test.out', { encoding: 'utf8' });
    expect(out).toMatchSnapshot();
  });
});
