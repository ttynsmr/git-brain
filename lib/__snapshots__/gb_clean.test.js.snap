// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clean filter actual file file stream 1`] = `
"version https://git-lfs.github.com/spec/v1
oid sha256:40a9ef7b88be7becfc0f503a712ccc82db446943d9cce6dc684a67cbc4cd18d6
size 15
"
`;

exports[`clean filter actual file file stream 2`] = `
"version https://git-lfs.github.com/spec/v1
oid sha256:40a9ef7b88be7becfc0f503a712ccc82db446943d9cce6dc684a67cbc4cd18d6
size 15
"
`;
